checkov/secrets/plugins/detector_utils.py

```diff
@@ -185,7 +185,8 @@ def extract_from_string(pattern: dict[Pattern[str], int] | None, string: str) ->
     for value_regex, group_number in pattern.items():
         match = value_regex.search(string)
         if match:
-            matches |= {match.group(group_number)}
+            remove_n = -1 if match.group(group_number).endswith('\n') else None
+            matches |= {match.group(group_number)[:remove_n]}
```

Comment:
you can use `rstrip` to strip characters from the right side of the string
```suggestion
            matches |= {match.group(group_number).rstrip('\n')}
```